generator/src/main/java/com/algorand/sdkutils/listeners/JavaGenerator.java

```diff
@@ -1,4 +1,933 @@
+
+
+    // Indicates if any property is added to this model
+    // used for skipping models with no parameters.  
+    private boolean modelPropertyAdded;
+
+    private String className;
+
+    // property name is the key
+    private TreeMap<String, TypeDef> properties;
+
+    TreeMap<String, Set<String>> imports;
+
+
+    private JavaGenerator javagen;
+    private String folder;
+
+    public JavaModelWriter(JavaGenerator javagen, String folder) {
+        currentModelBuffer = null;
+        pendingOpenFile = false;
+        this.javagen = javagen;
+        this.folder = folder;
+    }
+
+    public void close () {
+
+        writeProperties(currentModelBuffer);
+        writeCompareMethod(className, currentModelBuffer, getSortedProperties());
+
+        JavaGenerator.append(currentModelBuffer, "}\n");
+
+        JavaGenerator.append(modelFileWriter, Tools.getImports(imports));
+        if (modelPropertyAdded) {
+            JavaGenerator.append(modelFileWriter, currentModelBuffer);
+        }
+        modelPropertyAdded = false;
+
+        if (modelFileWriter != null) {
+            JavaGenerator.closeFile(modelFileWriter);
+        }
+        modelFileWriter = null;
+    }
+
+    public void newProperty(TypeDef type) {
+        modelPropertyAdded = true;
+        properties.put(Tools.getCamelCase(type.propertyName, false), type);
+    }
+
+    // newModel can write into one file at a time.
+    // This is a limitation, but there is no need for more.
+    // At any time, there can be one currentModelBuffer, and one JavaModelWriter
+    public void newModel(StructDef sDef, String packageName) {
+
+        if (pendingOpenFile) {
+            this.close();
+            pendingOpenFile = false;            
+        }
+        this.imports = new TreeMap<String, Set<String>>();
+        this.properties = new TreeMap<String, TypeDef>();
+
+        if (modelFileWriter == null) {
+
+            className = Tools.getCamelCase(sDef.name, true);
+
+            modelFileWriter = JavaGenerator.newFile(className, folder);
+            JavaGenerator.append(modelFileWriter, "package " + packageName + ";\n\n");
+
+            Tools.addImport(imports, "java.util.Objects"); // used by Objects.deepEquals
+
+            Tools.addImport(imports, "com.algorand.algosdk.v2.client.common.PathResponse");
+            Tools.addImport(imports, "com.fasterxml.jackson.annotation.JsonProperty");
+
+            currentModelBuffer = new StringBuilder();
+            if (sDef.doc != null) {
+                JavaGenerator.append(currentModelBuffer, Tools.formatComment(sDef.doc, "", true));
+            }
+            JavaGenerator.append(currentModelBuffer, "public class " + className + " extends PathResponse {\n\n");
+        }
+
+        pendingOpenFile = true;
+        modelPropertyAdded = false;
+    }
+
+    // Returns an iterator in sorted order of the properties (json nodes). 
+    Iterator<Entry<String, TypeDef>> getSortedProperties() {
```

Comment:
This helper seems like an unnecessary abstraction. Either change the cache to an `ArrayList` and call `sort()` in here, or get rid of the helper.